### PR TITLE
draft of good, great, 10x programmers

### PR DESCRIPTION
**Pull request by dbgrandi**

### CODE REVIEW

File: source/posts/good_great_10x.markdown [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+title: Good, Great, and 10x
+date: 2015/8/17
+published: false
+---
+
+The other day, I found myself in a conversation about what made a person a great programmer, and figured this is something I should strive to understand. I’d like to think that I'm a good programmer, and understanding what the path to greatness looks like could help me walk it.
+
+So, over the next few days I asked those around me, *”What makes a good programmer?”*
+
+Answers often started out short: **”Patience”**, **”Perseverance”**, **”Creativity”**. While these are good answers, they are also traits that could be useful in all kinds of endeavors...parenting, for example.
```
Comment (orta) [inline source/posts/good_great_10x.markdown]: > endeavors...parenting, for example

Spacing, and maybe capitalisation

File: source/posts/good_great_10x.markdown [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+### "One who uses the proper metaphor, and strives to find the proper metaphor."
+
+This is great. It's a larger version of the problem I often run into of spending a rather large amount of time thinking about what to name a variable.
+
+When you spend the time to find or create the right metaphor for something you are building, you don’t have to explain it so much to those who will use it.
+
+Years ago, [Metaphors We Live By](http://www.amazon.com/Metaphors-We-Live-George-Lakoff/dp/0226468011) was recommended to me by [Venkatesh Rao](http://ribbonfarm.com). If you want more insight into how we use metaphors to share concepts with others (which is pretty important in programming) and how metaphor affects both our thinking and our language...I highly recommend reading this. In fact, I’ll be re-reading it soon.
+
+One thing that I’ve found is the problems that arise when somebody uses a metaphor that is **almost** appropriate for what they are building. If you have a class or variable that is essentially a `Foo` except for two common edge cases where it behaves like a `Bar` you will inevitably have somebody who uses it exclusively like a `Foo` and causes problems.
```
Comment (orta) [inline source/posts/good_great_10x.markdown]: > One problem I've seen is when somebody uses a metaphor that is **almost** appropriate for what they are building.

File: source/posts/good_great_10x.markdown [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+### What about the 10x programmer?
+
+Many have written supporting or denying the existence of the 10x programmer. I’m squarely in the ‘they do exist’ camp, but perhaps not in the traditional sense.
+
+I recently came across (where was this!!!) a mention that *“the 10x programmer is one who makes the 5 closest programmers twice as effective.”*
+
+My view of the 10x programmer is that it's not about writing 10x more code. We don't get paid to write lines of code any more than a mathematician gets paid to write variables in LateX documents. We both get paid to solve problems. Code is a side effect of solving problems.
+
+Ideally, the 10x developer writes less code...or at least spends their time more efficiently in the code that they write.
```
Comment (orta) [inline source/posts/good_great_10x.markdown]: The ellipsis here feels a bit wishy-washy

**Comment by dbgrandi** [comment]
Thanks, @orta. Much appreciated!

File: source/posts/good_great_10x.markdown [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Years ago, [Metaphors We Live By](http://www.amazon.com/Metaphors-We-Live-George-Lakoff/dp/0226468011) was recommended to me by [Venkatesh Rao](http://ribbonfarm.com). If you want more insight into how we use metaphors to share concepts with others (which is pretty important in programming) and how metaphor affects both our thinking and our language...I highly recommend reading this. In fact, I’ll be re-reading it soon.
+
+One problem I’ve seen is when somebody uses a metaphor that is **almost** appropriate for what they are building. If you have a class or variable that is essentially a `Foo` except for two common edge cases where it behaves like a `Bar` you will inevitably have somebody who uses it exclusively like a `Foo` and causes problems.
+
+### What about the 10x programmer?
+
+Many have written supporting or denying the existence of the 10x programmer. I’m squarely in the ‘they do exist’ camp, but perhaps not in the traditional sense.
+
+I recently came across (where was this!!!) a mention that *“the 10x programmer is one who makes the 5 closest programmers twice as effective.”*
```
Comment (orta) [inline source/posts/good_great_10x.markdown]: Talked about this in the office, in art there is then distinction between good, better and best with respect to someone's artworks. 

* A good one should look across a spectrum of all their artworks to see the progression
* The best is usually a combination of master of technique, a master of subject matter, and of enhancing the field in general.

To quote a friend verbatim: "if you're not enhancing the field, it's just code."

Comment (orta) [inline source/posts/good_great_10x.markdown]: She also referenced https://philosophynow.org/issues/70/The_Better-Best_Fallacy

however its pay-wall'd - and the other link seems to be a really heavy read: http://arthistoryresources.net/renaissance-art-theory-2014/vasari-preface-2.html

Comment (dbgrandi) [inline source/posts/good_great_10x.markdown]: That's a strong concept. `Enhancing the field in general` is one of those phrases that is beautifully open for interpretation.

If some master of technique and subject matter came along and created a pod called `Monad` that clarified and conveyed monads in a succinct way, that would enhance the field.

I'll see if I can work this in.

**Pull request merged**